### ISSUE TITLE
VUE/Svelte like syntax

### ISSUE THREAD

**Issue by lifeart** [ember-template-imports/ember-template-imports#20]
This issue created to discuss how <script/template/style> approach may look like

**Comment by lifeart** [comment]
I would like to see something like SFC (Single File Components), with different languages, divided by tags.

example:
```hbs
<script>
import Component from '@glimmer/component';
export default class MyComponent extends Component {
   a = "hello, World!"
}
</script>

<template>
  {{this.a}}
</template>
```

**Comment by lifeart** [comment]
to get basic editors support we could start with ".vue" file extension for this POC

**Comment by NullVoxPopuli** [comment]
This would opt folks out of multiple tiny components in the same file. We already don't have that today, but would we be ok with never having that?

**Comment by lifeart** [comment]
@NullVoxPopuli I think we could have tag-based component definition here and share scope per-file, like

```vue
<template for="ChildComponent">

</template>
<template>
  Root Component <ChildComponent />
</template>
```

**Comment by NullVoxPopuli** [comment]
🤔 I suppose that could work, but during authoring, we're back to having string keys, which we'd have to build custom tooling around. 
_but_, with this vue-like approach, we don't have to add support for eslint.

**Comment by gossi** [comment]
I collected my idea in this gist: https://gist.github.com/gossi/6b15f5749c8fecd54d1b6d57168108b7

**Comment by rwjblue** [comment]
For what it is worth, the idea is that this repo is going to be very permissive with the formats that it allows using. The requirements are that the format is properly documented (including all of the scenarios mentioned for the two current formats in the README), and that adding support for a format that is unused does not cause performance regressions for the others.

tldr; PR's welcome 😸

**Comment by NullVoxPopuli** [comment]
It should be possible to convert *.svelte files to strict-mode files
(from https://svelte.dev/docs )
```svelte
<script>
	let x = 0;
	let y = 0;

	function yPlusAValue(value) {
		return value + y;
	}

	$: total = yPlusAValue(x);
</script>

Total: {total}
<button on:click={() => x++}>
	Increment X
</button>

<button on:click={() => y++}>
	Increment Y
</button>
```

becomes:
```js
export default setComponentTemplate(
  precompileTemplate(`
  Total: {this.total}
<button {{on 'click' this.firstClick}}>
	Increment X
</button>

<button {{on 'click' this.secondClick}}>
	Increment Y
</button>
  `),
  class Foo {
    // we don't actually know that these have to be tracked, we'd have to assume.
    // maybe we could inspect the click handlers and see what properties are accessed
    @tracked y = 0;
    @tracked x = 0;
    
    // this is a getter, that seemed over complicated in the svelte example
    get total() {
       return y + x;
    }
    
    // can't determine action name from anything with any consistency
   firstClick = () => x++;
   secondClick = () => y++;
  }
);
```

_but_,
this syntax forbids defining multiple components in the same file. I wonder how that would be done. I'm new to svelte, but interested in exploring these ideas.

**Comment by NullVoxPopuli** [comment]
This is what the svelte folks are doing for multiple components in a file https://github.com/sveltejs/svelte/issues/2940#issuecomment-791968175

**Comment by fredericrous** [comment]
It's totally personal taste but I'm no fan of having the `<script>` tag everywhere. I like the format bellow

![image](https://user-images.githubusercontent.com/702227/114726429-4caa7880-9d35-11eb-9080-3fe8bdff6948.png)
**screenshot taken from [Readme](https://github.com/ember-template-imports/ember-template-imports/#using-template-tags-and-gjs-files)*

To get css per component I could see a `<style>` tag there, maybe before the `<template>`. idk how difficult it would be to implement nor if highlighting tools would work out of the box

**Comment by NullVoxPopuli** [comment]
Closing because we went with the details laid out in rfc 779, thanks for all the chattin'!
